test/Vault1Test.js

```diff
@@ -2,12 +2,150 @@ const { expect } = require("chai");
 const { ethers } = require("hardhat");
 
 describe("Vault 1", () => {
-	beforeEach(async () => {
-		Vault1 = await ethers.getContractFactory("Vault1");
-		[owner] = await ethers.getSigners();
 
-		vault1 = await Vault1.deploy();
+  // I can create variables here ex:
+  let owner;
+  let account1;
+  // let nVault2;
+
+  let w5eth = BigInt(5 * 10 ** 18);
+  // let w3eth = BigInt(3 * 10 ** 18);
+  let w10eth = BigInt(6 * 10 ** 18);
+
+  // const token = new ethers.Contract(address, abi, signer);
+
+	beforeEach("deploy contracts", async () => {  
+		const accounts = await ethers.getSigners();
+
+		owner = accounts[0];
+		account1 = accounts[1];
+
+    const Vault2 = await ethers.getContractFactory("Vault2");
+    vault2 = await Vault2.deploy();
+
+    await vault2.deployed();
+  
+		const Vault1 = await ethers.getContractFactory("Vault1");
+		vault1 = await Vault1.deploy(account1.address, vault2.address);
 
 		await vault1.deployed();
 	});
+
+	describe("Deposit", function () {
+		it("should revert when amount equals zero", async function () {
+			await expect(
+				vault1.connect(owner).Deposit(0)
+			).to.be.revertedWith("Zero tokens");
+		});
+
+		it("should revert when user do not have VAULT token", async function () {
```

Comment:
Best to have a revert string message and check that

Reply:
You're right. The intention was to use string error from ERC20 @OpenZeppelin implementation function _transfer() "ERC20: transfer amount exceeds balance", but I missed it.